client/genericclient/generic_stream_service.go

```diff
@@ -26,8 +26,7 @@ func StreamingServiceInfo(g generic.Generic) *serviceinfo.ServiceInfo {
 }
 
 func newClientStreamingServiceInfo(g generic.Generic) *serviceinfo.ServiceInfo {
-	readerWriter := g.MessageReaderWriter()
-	if readerWriter == nil {
+	if g.PayloadCodec() != nil {
```

Comment:
Why change this judgement? These two are different at their meaning.

Reply:
When PayloadCodec is not nil, MessageReaderWriter is nil, right?
Users, who customize their own generic, implement PayloadCodec when they have different codec between reader and writer or when they use binary generic. So, I thought judging PayloadCodec here actually didn't change the logic meaning.
Plus, I thought PayloadCodec cost less than calling MessageReaderWriter.